tsdb/index/tsi1/partition.go

```diff
@@ -1401,9 +1401,35 @@ func (m *Manifest) Write() (int64, error) {
 	}
 	buf = append(buf, '\n')
 
-	if err := os.WriteFile(m.path, buf, 0666); err != nil {
+	f, err := os.CreateTemp(filepath.Dir(m.path), ManifestFileName)
```

Comment:
Usually we just add `.tmp` to the end for our tmp files, rather than `os.CreateTemp`. Is there a reason to break the pattern here?

Reply:
@lesam - Just a fan of using package functionality over home-grown.